### ISSUE TITLE
Wordpress support

### ISSUE THREAD

**Issue by creator2earth** [danielgindi/jquery-backstretch#24]
Literally spent 5hrs trying to just make this plugin work, for some reason I just cant get this to work. Normally I move on, but this plugin could mean about 50% of the site I'm working on done. 
This is what I've tried so far and so far none have worked
- wp_enqueue_script
- wp_register_script
  -wp_localize
- placed the script and right above body tag/inside header/alone in separate JS file.
- Setting it to a div $(".jumbotron").backstretch("http://placekitten.com/200/300");
- Setting it to the body $.backstretch('.....);
  -various other ways of making it stick with php
- tried countless divs
  -deregister/registered by jquery in function.php
- putting <?php wp_enqueue('jquery') ?> right before the wp_head
-  linking to image path in my folder
- linking in dropbox
- lots of stackoverflow
  -changing certain lines in the plugin its self.
- downloaded a plugin that bumps me up to the latest version of jquery via the wordpress admin. 

Is there a step by step, bulletproof way to make this work on WordPress? This would benefit a lot of people I think. I will settle for the smallest image right now, can share my whole theme file if I have to. I suspect the issue is loading the jquery but wordPress migrates it's jquery plus add another file on top of that, plus whatever the jquery the plugin needs. The source code gets aggressive after a while.

**Comment by danielgindi** [comment]
Actually it only required jQuery. That's it.
Did you try to simply run code from our examples?
There shouldn't be any problem specifically with Wordpress. You just have to let it know that backstretch depends on jQuery so it will spit out the <script> tags in the right order

**Comment by creator2earth** [comment]
Hey Daniel, I followed the examples(which I did not bother looking at before for some reason), and still the same thing. I don't know if I am doing something wrong or  am missing something. I really suspect it's something so simple that I am missing. Is there anyway I can send you a private dropbox link with all my files so you can help me troubleshoot this. Like I said before, I usually move on but really want to use this plugin. I am also getting this message saying "backstretch is not a function"... If you can help me with this that would be stellar. Thanks.

**Comment by danives3d** [comment]
I am going to assume that, because you're using Wordpress, you need to replace the dollar sign with `jQuery`, so the instantiation of it would be `jQuery.backstretch()` as opposed to `$.backstretch`

I mean you don't _have_ to, depending on if you use noConflict or something similar to enable the use of the dollar sign, but that might be your issue.

**Comment by creator2earth** [comment]
No, I'm already using jQuery no conflict. This is crazy, can someone just look at my source code? I feel like this would be the only way to narrow this down. It might take someone 2 secs to look at my code and tell me what I'm doing wrong vs. this back and forth. Maybe backsretch isn't for me, but that just annoyed me talking.

**Comment by creator2earth** [comment]
I even tried inserting my image thru css using background: url() method and then try to target the div using backstretch and still nothing............?

**Comment by creator2earth** [comment]
Hey you can close this issue if you want now. I figured it out and I'm using the forked version that lets you add backstretch to any div! Thanks for the help attempt!

**Comment by creator2earth** [comment]
Hey, I reverted back to the original plugin that you forked, because the other repo isn't really being maintained. I'm currently using wp_query to pull in the right content and Im storing the image attachment as a variable. I then went and opened up a script tag and assign my original php variable to a javascript variable using json_encode($main_thumb) so it looks something like this:
var myVar = <?php echo json_encode($main_thumb) ?>.
 I then went ahead in a file called theme.js which is enqueue and added this code(mind you the div I'm targeting has a class named "test":

 jQuery(document).ready(function($) {

   $('.test').backstretch(myVar);

```
  });//End No conflict
```

After saving and refreshing my browser, nothing shows up. But when I go into the Dev Tools(Chrome), I find the image has indeed loaded, the text comes up as red, so I'm guessing there is an error , but no console error shows. I then go further and try to investigate by opening up the source code, I see that that both the image and backstretch.min.js is in fact getting loaded in the footer as I instructed in my functions.php. 

Mind you, if I take off the ".test" and just use backstretch with the variable it does indeed fill the whole body with the picture I want, but as soon as I try to add it to a div element nothing shows up, showing no error in the console. 

Sorry that was so long, but is there a better way to do this using php, jQuery and WordPress, because this method is not working and is causing me much grief.

Thanks, hopefully I explained this good enough for someone to help me with this issue.

Also this is what I'm getting in the source code as far as WordPress pulling in the image, while using json_encode as I try to put it in a variable:

var myVar = "http:\/\/localhost\/sheyayoga.com\/wp-content\/uploads\/2016\/07\/sheya_pose.jpg";

**Comment by neundex** [comment]
Here is a possible workflow:

In your **functions.php**:

``` php
function backstretch_json( $slide_id ) {

  $slide = get_post_thumbnail_id( $slide_id ); //supposing your are using the featured image
  $json = [];

  if( $slide ) {
    $json[] = wp_get_attachment_image_src( $slide, 'full' )[0]; //use your custom image size instead of full
  }

  echo json_encode( $json );

}
```

In your **markup:**

``` html
<div class="backstretch" data-slides='<?php backstretch_json( $post->ID ); ?>'></div>
```

In your **js:**

``` js
var slides = $('.backstretch').data('slides');
$('.backstretch').backstretch(slides);
```

In your **css:**

``` css
.backstretch {
  min-height: 100vh; /*use the height you need*/ 
}
```

**Comment by creator2earth** [comment]
Hey thanks @neundex  that worked to perfection!

**Comment by creator2earth** [comment]
@neundex Hey so I got my main jumbotron image working. How would this be applied to multiple images on the page. I checked the source code when I tried to apply the same concept to another image and it comes up on the source code but the image doesn't show up on the site, and also note none of the backstretch classes are applied. I went back and even tried messing with the CSS and nothing. Do I have to go back to functions.php and do anything differently or make up another function for each picture?

Let me know if I need to explain this better.
